### ISSUE TITLE
Migrate android_semantics_testing to null safety

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter#90633]
This was attempted a couple of times as part of the wider null safety migration project (#84014). 

The package in question is: ./dev/integration_tests/android_semantics_testing/pubspec.yaml 

The attempts (#84136, #84156, and #90466) passed all the tests locally and during pre-submit, but would fail when it hit the device lab. We need to investigate why and resolve the issue.

**Comment by darrenaustin** [comment]
cc @goderbauer

**Comment by goderbauer** [comment]
With the imminent down-turn off nun-null-safe code, we need to attempt this migration again and - if it fails again - figure out the reason.

/cc @chunhtai

**Comment by chunhtai** [comment]
is there an error message for the failure?

**Comment by goderbauer** [comment]
One of the revert PRs lists the error message: https://github.com/flutter/flutter/pull/84154

Might be worthwhile to check whether this is still happening, though, since that was over a year ago)

**Comment by mit-mit** [comment]
@chunhtai were you able to try this again?
